dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/RouteController.java

```diff
@@ -0,0 +1,96 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller;
+
+import java.io.IOException;
+
+import javax.servlet.http.HttpServletRequest;
+
+import lombok.RequiredArgsConstructor;
+
+import org.hisp.dhis.common.DhisApiVersion;
+import org.hisp.dhis.common.OpenApi;
+import org.hisp.dhis.route.Route;
+import org.hisp.dhis.route.RouteService;
+import org.hisp.dhis.schema.descriptors.RouteSchemaDescriptor;
+import org.hisp.dhis.user.CurrentUser;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.webapi.mvc.annotation.ApiVersion;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+
+/**
+ * @author Morten Olav Hansen
+ */
+@RestController
+@OpenApi.Tags( "integration" )
+@RequiredArgsConstructor
+@RequestMapping( value = RouteSchemaDescriptor.API_ENDPOINT )
+@ApiVersion( { DhisApiVersion.DEFAULT, DhisApiVersion.ALL } )
+public class RouteController
+    extends AbstractCrudController<Route>
+{
+    private final RouteService routeService;
+
+    @RequestMapping( value = "/{id}/run", method = { RequestMethod.GET, RequestMethod.POST } )
+    public ResponseEntity<String> run( @PathVariable( "id" ) String id, @CurrentUser User user,
+        HttpServletRequest request )
+        throws IOException
+    {
+        Route route = routeService.getDecryptedRoute( id );
+
+        if ( route == null )
+        {
+            throw new HttpClientErrorException( HttpStatus.NOT_FOUND, String.format( "Route %s not found", id ) );
+        }
+
+        if ( !aclService.canRead( user, route ) && !user.hasAnyAuthority( route.getAuthorities() ) )
+        {
+            throw new HttpClientErrorException( HttpStatus.FORBIDDEN, "User not authorized" );
+        }
+
+        ResponseEntity<String> entity = routeService.exec( route, request );
+
+        if ( entity.getStatusCode().is4xxClientError() )
+        {
+            throw new HttpClientErrorException( entity.getStatusCode() );
```

Comment:
I guess in line with some other discussions we would make these into `feedback.ConflictException` with a custom `ErrorCode` for this particular issue. Than you can have a message where the status code is added.